src/libs/sys-util-rs/src/numa.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) 2021 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::collections::HashMap;
+use std::fs::DirEntry;
+use std::io::Read;
+use std::path::Path;
+
+use lazy_static::lazy_static;
+
+use crate::cpu::split_cpus;
+use crate::sl;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Invalid CPU number {0}")]
+    InvalidCpu(u32),
+    #[error("Invalid node file name {0}")]
+    InvalidNodeFileName(String),
+    #[error("Can not read from file {0}, {1:?}")]
+    ReadFile(String, #[source] std::io::Error),
+    #[error("Can not open from file {0}, {1:?}")]
+    OpenFile(String, #[source] std::io::Error),
+    #[error("Can not parse CPU info, {0:?}")]
+    ParseCpuInfo(#[from] crate::cpu::Error),
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+// global config in UT
+#[cfg(test)]
+lazy_static! {
+    static ref SYS_FS_PREFIX: String = format!(
+        "{}/test/texture",
+        std::env::var("CARGO_MANIFEST_DIR").unwrap()
+    );
+}
+
+// global config in release
+#[cfg(not(test))]
+lazy_static! {
+    static ref SYS_FS_PREFIX: String = String::new();
+}
+
+lazy_static! {
+    // numa node file for UT, we can mock data
+    static ref NUMA_NODE_PATH: String = format!("{}/sys/devices/system/node", &*SYS_FS_PREFIX);
```

Comment:
Using `Path` and `Path::join` is probably better than `String` and `format!`.

Reply:
fixed